builtin/credential/aws/path_config_certificate.go

```diff
@@ -0,0 +1,245 @@
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		// Existence check depends on this being nil when the storage entry is not present.
+		return nil, nil
+	}
+
+	var result awsPublicCert
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+// pathConfigCertificateRead is used to view the configured AWS Public Key that is
+// used to verify the PKCS#7 signature of the instance identity document.
+func (b *backend) pathConfigCertificateRead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	certificateEntry, err := awsPublicCertificateEntry(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if certificateEntry == nil {
+		return nil, nil
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"aws_public_cert": certificateEntry.AWSPublicCert,
+		},
+	}, nil
+}
+
+// pathConfigCertificateCreateUpdate is used to register an AWS Public Key that is
+// used to verify the PKCS#7 signature of the instance identity document.
+func (b *backend) pathConfigCertificateCreateUpdate(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+
+	// Check if there is already a certificate entry registered.
+	certEntry, err := awsPublicCertificateEntry(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if certEntry == nil {
+		certEntry = &awsPublicCert{}
+	}
+
+	// Check if the value is provided by the client.
+	certStrB64, ok := data.GetOk("aws_public_cert")
+	if ok {
+		certBytes, err := base64.StdEncoding.DecodeString(certStrB64.(string))
+		if err != nil {
+			return nil, err
+		}
+
+		certEntry.AWSPublicCert = string(certBytes)
+	} else if req.Operation == logical.CreateOperation {
+		certEntry.AWSPublicCert = data.Get("aws_public_cert").(string)
+	}
+
+	// If explicitly set to empty string, error out.
+	if certEntry.AWSPublicCert == "" {
+		return logical.ErrorResponse("missing aws_public_cert"), nil
+	}
+
+	// Verify the certificate by decoding it and parsing it.
+	publicCert, err := decodePEMAndParseCertificate(certEntry.AWSPublicCert)
+	if err != nil {
+		return nil, err
+	}
+	if publicCert == nil {
+		return logical.ErrorResponse("invalid certificate; failed to decode and parse certificate"), nil
+	}
+
+	// Before trusting the signature provided, validate its signature.
+
+	// Extract the signature of the certificate.
+	dsaSig := &dsaSignature{}
+	dsaSigRest, err := asn1.Unmarshal(publicCert.Signature, dsaSig)
+	if err != nil {
+		return nil, err
+	}
+	if len(dsaSigRest) != 0 {
+		return nil, fmt.Errorf("failed to unmarshal certificate's signature")
+	}
+
+	certHashFunc := crypto.SHA1.New()
+
+	// RawTBSCertificate will contain the information in the certificate that is signed.
+	certHashFunc.Write(publicCert.RawTBSCertificate)
+
+	// Verify the signature using the public key present in the certificate.
+	if !dsa.Verify(publicCert.PublicKey.(*dsa.PublicKey), certHashFunc.Sum(nil), dsaSig.R, dsaSig.S) {
```

Comment:
@vishalnayak might be able to shed light on the purpose there, but I think you are probably correct here.